editor.planx.uk/src/pages/FlowEditor/lib/store/navigation.ts

```diff
@@ -21,10 +29,11 @@ export interface NavigationStore {
   initNavigationStore: () => void;
   updateSectionData: () => void;
   filterFlowByType: (type: TYPES) => Store.flow;
+  sectionStatuses: () => Record<string, SectionStatus>;
 }
 
 export const navigationStore: StateCreator<
-  NavigationStore & SharedStore,
+  NavigationStore & SharedStore & PreviewStore,
```

Comment:
adding the whole `PreviewStore` here feels a bit wrong here, but I want to be able to get `currentCard` & `upcomingCardIds` to calculate the section status. should be able to eliminate this once i'm calculating section status based on breadcrumbs instead.

Reply:
Does this suggest that `currntCard` and `upcomingCardIds` (perhaps also others) should be extracted to a new `LogicStore` (or something similar) to keep the responsibility of the store focused and clear?

Reply:
yep, smaller stores going forward makes sense to me - but let's queue that refactor up later - because as Daf hints at, I think it'll be quite the exercise to effectively categorise & split up what we currently have (because so many methods call each other, therefore may still be tricky at times to avoid imports like this, etc).